libbeat/mapping/field.go

```diff
@@ -119,6 +122,55 @@ func (f *Field) Validate() error {
 	return nil
 }
 
+func (f *Field) validateType() error {
+	switch strings.ToLower(f.Type) {
+	case "text", "keyword":
+		return errors.Wrapf(stringType.validate(f.Format), "field %s", f.Name)
+	case "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float":
+		return errors.Wrapf(numberType.validate(f.Format), "field %s", f.Name)
+	case "date", "date_nanos":
+		return errors.Wrapf(dateType.validate(f.Format), "field %s", f.Name)
+	case "geo_point":
+		return errors.Wrapf(geoPointType.validate(f.Format), "field %s", f.Name)
+	case "boolean", "binary", "ip", "alias", "array":
+		if f.Format != "" {
+			return fmt.Errorf("no format expected for field %s, found: %s", f.Name, f.Format)
+		}
+	case "object", "group", "nested":
+		// No check for them yet
+	case "":
+		// Module keys, not used as fields
+	default:
+		// There are more types, not being used by beats, to be added if needed
+		return fmt.Errorf("unexpected type '%s' for field '%s'", f.Type, f.Name)
+	}
+	return nil
+}
+
+type fieldTypeGroup struct {
+	name    string
+	formats []string
+}
+
+var (
+	stringType   = fieldTypeGroup{"string", []string{"string", "url"}}
+	numberType   = fieldTypeGroup{"number", []string{"string", "url", "bytes", "duration", "number", "percent", "color"}}
```

Comment:
I think we should start writing constants instead of anonymous string literals. This way we can avoid weird typos, add some comment to know what they are, improve readability.

In this particular case it will help to clearly see (I mean literally see with the eyes :smile: ) what is a type (common, complex, geo or specialized) and what is a format of an index pattern.

It's just a suggestion, no blocking.

Reply:
I though about that, but I had to define loads of constants that actually are never going to be used apart of for these lines.
But, I have added a comment explaining where these values come from so it is more clear when someone has to revisit this code.